src/ressources/commonFields/FullAddressField.tsx

```diff
@@ -0,0 +1,11 @@
+import { useRecordContext, UseRecordContextParams } from "react-admin";
+import { Typography } from "@mui/material";
+export const FullAddressField = (props: UseRecordContextParams) => {
+  const record = useRecordContext(props);
```

Comment:
Protip: you can actually strong type the return value of the `useRecordContext` hook, like this:

```tsx
// in customer.ts
export interface Customer {
  id: number;
  [...]
  address: string;
  zipcode: string;
  city: string;
}

// in FullAddressField.tsx
const record = useRecordContext<Customer>(props); // record will be of type Customer and have Intellisense
```